propagators/opentelemetry-propagator-opentracing/src/OpenTracingPropagator.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Baggage,
+  Context,
+  getBaggage,
+  getSpanContext,
+  isSpanContextValid,
+  isValidSpanId,
+  isValidTraceId,
+  setBaggage,
+  setSpanContext,
+  TextMapGetter,
+  TextMapPropagator,
+  TextMapSetter,
+  TraceFlags,
+} from '@opentelemetry/api';
+
+/** OT header keys */
+export const OT_TRACE_ID_HEADER = 'ot-tracer-traceid';
+export const OT_SPAN_ID_HEADER = 'ot-tracer-spanid';
+export const OT_SAMPLED_HEADER = 'ot-tracer-sampled';
+export const OT_BAGGAGE_PREFIX = 'ot-baggage-';
+
+const FIELDS = [OT_TRACE_ID_HEADER, OT_SPAN_ID_HEADER, OT_SAMPLED_HEADER];
```

Comment:
The multi-header baggage breaks the fields usecase. Clearing fields out of an object won't work if we don't know all the fields that were used. Saving used fields might result in a memory leak. Not sure how to resolve this. It should at least be documented.

Reply:
I added a comment further down above the `fields()` method.

Reply:
I think this should be also mentioned in readme similar as the restrictions regarding traceId which are already there.

Reply:
can we do opposite then, keep the names of fields that should be there and clear all the rest that are not on that list?

Reply:
I don't think that we actually use the `fields` method in the API or SDK, but it is there for users or instrumentation authors to use. I documented the special case on the readme. The specification seems to hint at this possibility as well:

> Observe that some Propagators may define, besides the returned values, additional fields with variable names. To get a full list of fields for a specific carrier object, use the Keys operation.

From https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/api-propagators.md#fields